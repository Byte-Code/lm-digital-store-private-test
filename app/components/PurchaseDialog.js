import React, { Component, PropTypes } from 'react';
import QRCode from 'qrcode.react';
import styled from 'styled-components';
import Dialog from 'material-ui/Dialog';

import CloseButton from './CloseButton';

export const Wrapper = styled.div`
`;

const Header = styled.div`
  display: flex;
  justify-content: center;
  margin-bottom: 114px;
  &>p {
    font-size: 14px;
    text-align: center;
    color: #67cb33;
    border-bottom: 2px solid #67cb33;
    padding-bottom: 9px;
    text-transform: uppercase;
  }
`;

const Title = styled.h1`
  font-size: 48px;
  text-align: center;
  margin-bottom: 13px;
`;

const Subtitle = styled.p`
  font-size: 14px;
  font-family: LeroyMerlinSans Light-Italic;
  text-align: center;
  margin-bottom: 57px;
`;

const Content = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

export default class PurchaseDialog extends Component {
  static propTypes = {
    productCode: PropTypes.string.isRequired,
    productSlug: PropTypes.string.isRequired,
    children: PropTypes.node.isRequired
  }

  constructor(props) {
    super(props);
    this.state = {
      open: false
    };
  }

  handleOpen = () => {
    this.setState({ open: true });
  }

  handleClose = () => {
    this.setState({ open: false });
  }

  render() {
    const { children, productCode, productSlug } = this.props;
    const url = `https://www.leroymerlin.it/catalogo/${productSlug}-${productCode}-p`;

    return (
      <Wrapper onClick={this.handleOpen}>
        {children}
        <Dialog
          modal={false}
          onRequestClose={this.handleClose}
          open={this.state.open}
          contentStyle={{ width: '100%' }}
          bodyStyle={{ padding: 75, background: '#fff' }}
        >
          <CloseButton handleClick={this.handleClose} backgroundColor="#fff" />
          <Header>
            <p>genera qr-code</p>
          </Header>
          <Title>Genera QR-Code</Title>
          <Subtitle>
            Inquadra il QR-Code con il tuo smartphone per aprire il link alla scheda prodotto.
          </Subtitle>
          <Content>
            <QRCode
              value={url}
              size={200}
            />
          </Content>
        </Dialog>
      </Wrapper>
    );
  }
}
