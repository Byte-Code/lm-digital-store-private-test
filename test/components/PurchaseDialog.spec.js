import React from 'react';
import { shallow } from 'enzyme';

import PurchaseDialog, { Wrapper } from '../../app/components/PurchaseDialog';

const productCode = 'PROD123';
const productSlug = 'SLUG123';

describe('PurchaseDialog', () => {
  const result = shallow(
    <PurchaseDialog
      productCode={productCode}
      productSlug={productSlug}
    >
      <p>children</p>
    </PurchaseDialog>
  );

  it('should initially render with a closed dialog', () => {
    expect(result.state('open')).toBe(false);
    expect(result).toMatchSnapshot();
  });

  it('should open the dialog once clicked', () => {
    result.find(Wrapper).simulate('click');
    expect(result.state('open')).toBe(true);
    expect(result).toMatchSnapshot();
  });
});
