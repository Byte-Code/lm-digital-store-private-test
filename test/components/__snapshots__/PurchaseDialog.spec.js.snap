// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PurchaseDialog should initially render with a closed dialog 1`] = `
<styled.div
  onClick={[Function]}
>
  <p>
    children
  </p>
  <Dialog
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    bodyStyle={
      Object {
        "background": "#fff",
        "padding": 75,
      }
    }
    contentStyle={
      Object {
        "width": "100%",
      }
    }
    modal={false}
    onRequestClose={[Function]}
    open={false}
    repositionOnUpdate={true}
  >
    <CloseButton
      backgroundColor="#fff"
      fill="#858585"
      handleClick={[Function]}
      right={0}
      top={0}
    />
    <styled.div>
      <p>
        genera qr-code
      </p>
    </styled.div>
    <styled.h1>
      Genera QR-Code
    </styled.h1>
    <styled.p>
      Inquadra il QR-Code con il tuo smartphone per aprire il link alla scheda prodotto.
    </styled.p>
    <styled.div>
      <QRCode
        bgColor="#FFFFFF"
        fgColor="#000000"
        level="L"
        size={200}
        value="https://www.leroymerlin.it/catalogo/SLUG123-PROD123-p"
      />
    </styled.div>
  </Dialog>
</styled.div>
`;

exports[`PurchaseDialog should open the dialog once clicked 1`] = `
<styled.div
  onClick={[Function]}
>
  <p>
    children
  </p>
  <Dialog
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    bodyStyle={
      Object {
        "background": "#fff",
        "padding": 75,
      }
    }
    contentStyle={
      Object {
        "width": "100%",
      }
    }
    modal={false}
    onRequestClose={[Function]}
    open={true}
    repositionOnUpdate={true}
  >
    <CloseButton
      backgroundColor="#fff"
      fill="#858585"
      handleClick={[Function]}
      right={0}
      top={0}
    />
    <styled.div>
      <p>
        genera qr-code
      </p>
    </styled.div>
    <styled.h1>
      Genera QR-Code
    </styled.h1>
    <styled.p>
      Inquadra il QR-Code con il tuo smartphone per aprire il link alla scheda prodotto.
    </styled.p>
    <styled.div>
      <QRCode
        bgColor="#FFFFFF"
        fgColor="#000000"
        level="L"
        size={200}
        value="https://www.leroymerlin.it/catalogo/SLUG123-PROD123-p"
      />
    </styled.div>
  </Dialog>
</styled.div>
`;
